### PR TITLE
fix: type error of TypeScript

### PR DESCRIPTION
**Pull request by fi3ework**
Thanks for taking the effort to create a PR!

If you are creating an extensive PR, you might want to open an issue with your idea first, so that you don't put a lot of effort in an PR that wouldn't be accepted. Please prepend pull requests with `WIP: ` if they are not yet finished
PR checklist:

* [ ] Added unit tests
* [ ] Updated changelog
* [ ] Updated docs (either in the description of this PR as markdown, or as separate PR on the `gh-pages` branch. Please refer to this PR). For new functionality, at least [API.md](https://github.com/mobxjs/mobx/blob/gh-pages/docs/refguide/api.md) should be updated
* [ ] Added typescript typings
* [x] Verified that there is no significant performance drop (`npm run perf`)

Run `yarn run quick-build` will fail because of some type errors. This PR fix them and quick-build works fine now 🤩

### CODE REVIEW

File: src/types/observablemap.ts [hunk 1]
```diff
@@ -306,6 +306,7 @@ export class ObservableMap<K = any, V = any>
                 other.forEach((value, key) => this.set(key, value))
             } else if (other !== null && other !== undefined)
                 fail("Cannot initialize map from " + other)
+            return
```
Comment (mweststrate) [inline src/types/observablemap.ts]: NIT: I think dropping `return` here and on line 305 woudl achieve the same. (`fail` always throws)

**Comment by mweststrate** [comment]
Thanks! Looking good

File: src/types/observablemap.ts (hunk 1 above)
Comment (fi3ework) [reply src/types/observablemap.ts]: Agreed. This would be better as there is nothing need to be returned 😀

**Comment by mweststrate** [comment]
Thanks!

**Pull request merged**